runtime/server/mcp_client.go

```diff
@@ -0,0 +1,91 @@
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/mark3labs/mcp-go/client"
+	"github.com/mark3labs/mcp-go/mcp"
+	"github.com/mark3labs/mcp-go/server"
+	"github.com/rilldata/rill/runtime"
+)
+
+func (s *Server) newMCPClient(mcpServer *server.MCPServer) *client.Client {
+	client, err := client.NewInProcessClient(mcpServer)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
Return the error instead (panics generally should not be used) – applies a few places in this file, some of them it might just be a `ctx` error like timeout.